ImageFeed/Services/ImagesListService.swift

```diff
@@ -0,0 +1,197 @@
+        case welcomeDescription = "description"
+        case isLiked = "liked_by_user"
+        case urls = "urls"
+        case width = "width"
+        case height = "height"
+    }
+}
+
+struct UrlsResult: Decodable {
+    let thumbImageURL: String?
+    let largeImageURL: String?
+    
+    enum CodingKeys: String, CodingKey {
+        case thumbImageURL = "thumb"
+        case largeImageURL = "full"
+    }
+}
+
+
+final class ImagesListService {
+    private (set) var photos: [Photo] = []
+    private var lastLoadedPage: Int?
+    private let perPage = "10"
+    private var task: URLSessionTask?
+    static let DidChangeNotification = Notification.Name(rawValue: "ImagesListServiceDidChange")
+    static let shared = ImagesListService()
+    private let storageToken = OAuth2TokenStorage()
+    private let dateFormatter = ISO8601DateFormatter()
+    
+    func fetchPhotosNextPage() {
+        assert(Thread.isMainThread)
+        task?.cancel()
+        
+        let page = lastLoadedPage == nil
+        ? 1
+        : lastLoadedPage! + 1
+        guard let request = photoRequest(page: String(page), perPage: perPage) else { return }
+        let task = URLSession.shared.objectTask(for: request) { [weak self] (result: Result<[PhotoResult], Error>) in
+            DispatchQueue.main.async {
+                guard let self else { return }
+                self.task = nil
+                switch result {
+                case .success(let photoResults):
+                    for photoResult in photoResults {
+                        self.photos.append(self.convert(photoResult))
+                    }
+                    self.lastLoadedPage = page
+                    NotificationCenter.default
+                        .post(
+                            name: ImagesListService.DidChangeNotification,
+                            object: self,
+                            userInfo: ["Images" : self.photos])
+                case .failure(_):
+                    break
+                }
+            }
+        }
+        self.task = task
+        task.resume()
+    }
+    
+    private func photoRequest(page: String, perPage: String) -> URLRequest? {
+        guard let url = URL(string: "https://api.unsplash.com") else { return nil }
+        var request = URLRequest.makeHTTPRequest(
+            path: "/photos?page=\(page)&&per_page=\(perPage)",
+            httpMethod: "GET",
+            baseURL: url)
+        if let token = OAuth2TokenStorage().token {
+            request.setValue("Bearer \(token)", forHTTPHeaderField: "Authorization")
+        }
+        return request
+    }
+    
+    private func convert(_ photoResult: PhotoResult) -> Photo {
+        let dateFormatter = DateFormatter()
+        dateFormatter.locale = Locale(identifier: "en_US_POSIX")
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZ"
+        let _ = dateFormatter.date(from: photoResult.createdAt ?? "")
+        
+        return Photo.init(id: photoResult.id,
```

Comment:
в swift инициализаторы доступны по имени типа: 
```suggestion
        return Photo(id: photoResult.id,
```